spark3/src/main/java/org/apache/iceberg/spark/actions/compaction/SparkBinningCompactionStrategy.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.actions.compaction;
+
+import java.util.List;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.spark.source.SourceUtil;
+import org.apache.spark.sql.SparkSession;
+
+public class SparkBinningCompactionStrategy extends BinningCompactionStrategy {
```

Comment:
Here , we've introduced a `SparkBinningCompactionStrategy` ,   that sounds like is customized for the spark query engine.  I think the compaction strategy is common to various query engines, it just use its own strategy to plan the sub-tasks for each job,  the real job execution is binded to query engine.  Maybe we will need to move `rewriteFiles` out of the strategy intefaces.

Reply:
I was trying to think of a way to do this, but it's difficult because the actual job execution is also algorithm specific. It is both Framework (Spark) and Strategy (binning) specific. Rewriting files by grouping them is one example, but the algorithm could be rewriting files by sorting them.